src/server.rs

```diff
@@ -89,6 +90,55 @@ pub extern "C" fn rustls_server_config_builder_from_config(
     }
 }
 
+/// Set the TLS protocol versions to use when negotiating a TLS session.
+///
+/// `tls_version` is the version of the protocol, as defined in rfc8446,
+/// ch. 4.2.1 and end of ch. 5.1. Some values are defined in
+/// `rustls_tls_version` for convenience.
+///
+/// `versions` will only be used during the call and the application retains
+/// ownership. `len` is the number of consecutive `ui16` pointed to by `versions`.
+#[no_mangle]
+pub extern "C" fn rustls_server_config_builder_set_versions(
+    builder: *mut rustls_server_config_builder,
+    tls_versions: *const u16,
+    len: size_t,
+) -> rustls_result {
+    ffi_panic_boundary! {
+        let config: &mut ServerConfig = try_ref_from_ptr!(builder, &mut ServerConfig);
+        config.versions.clear();
+        unsafe {
+            let x: &[u16] = slice::from_raw_parts(tls_versions, len);
+            for i in x {
+                match rustls_tls_version_from_u16(*i) {
+                    Some(pversion) => config.versions.push(pversion),
+                    None => return rustls_result::AlertIllegalParameter
```

Comment:
The `Alert` values in `rustls_result` are meant for conveying a TLS alert. I'd prefer to allocate a new `rustls_result` value rather than use this one outside its original purpose.

Reply:
Changed. Added rustls_result::InvalidParameter.